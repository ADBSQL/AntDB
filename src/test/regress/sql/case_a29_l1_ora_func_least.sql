set grammar to oracle;
select least(1,2,3,4) from dual;
select least(1.1,to_number(1.1),exp(2)) from dual;
select least(5.2,to_number(1.1),to_char(1.1)) from dual;
select least(5.2,to_number(1.1),'1') from dual;
select least(to_char(100),to_char(12),to_char(3.1)) from dual;
select least(to_number(3),'100','12') from dual;
select least('100',to_number(3),12) from dual;
select least('100','12',to_number(3)) from dual;
select least('100','12',exp(2)) from dual;
select least('abc','ab','abcd') from dual;
select least('abc','ab','1') from dual;
--select least('你好','加油') from dual;
select least('','a') from dual;
select least('a',null) from dual;
select least(null,'') from dual;

create table t4test(id int,txt varchar);
insert into t4test values(1,null);
insert into t4test values(2,'ase');
insert into t4test values(3,least('aa','12','cc'));
select * from t4test order by id;
select least(txt,'sds','ab') from t4test order by id;
update t4test set txt=least(txt,'aba','ads');
select * from t4test order by id;
drop table t4test;
