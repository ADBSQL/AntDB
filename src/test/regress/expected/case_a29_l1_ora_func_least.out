set grammar to oracle;
select least(1,2,3,4) from dual;
 LEAST 
-------
     1
(1 row)

select least(1.1,to_number(1.1),exp(2)) from dual;
 LEAST 
-------
   1.1
(1 row)

select least(5.2,to_number(1.1),to_char(1.1)) from dual;
 LEAST 
-------
   1.1
(1 row)

select least(5.2,to_number(1.1),'1') from dual;
 LEAST 
-------
     1
(1 row)

select least(to_char(100),to_char(12),to_char(3.1)) from dual;
 LEAST 
-------
 100
(1 row)

select least(to_number(3),'100','12') from dual;
 LEAST 
-------
     3
(1 row)

select least('100',to_number(3),12) from dual;
 LEAST 
-------
 100
(1 row)

select least('100','12',to_number(3)) from dual;
 LEAST 
-------
 100
(1 row)

select least('100','12',exp(2)) from dual;
 LEAST 
-------
 100
(1 row)

select least('abc','ab','abcd') from dual;
 LEAST 
-------
 ab
(1 row)

select least('abc','ab','1') from dual;
 LEAST 
-------
 1
(1 row)

--select least('你好','加油') from dual;
select least('','a') from dual;
 LEAST 
-------
 
(1 row)

select least('a',null) from dual;
 LEAST 
-------
 
(1 row)

select least(null,'') from dual;
 LEAST 
-------
 
(1 row)

create table t4test(id int,txt varchar);
insert into t4test values(1,null);
insert into t4test values(2,'ase');
insert into t4test values(3,least('aa','12','cc'));
select * from t4test order by id;
 ID | TXT 
----+-----
  1 | 
  2 | ase
  3 | 12
(3 rows)

select least(txt,'sds','ab') from t4test order by id;
 LEAST 
-------
 ab
 ab
 12
(3 rows)

update t4test set txt=least(txt,'aba','ads');
select * from t4test order by id;
 ID | TXT 
----+-----
  1 | aba
  2 | aba
  3 | 12
(3 rows)

drop table t4test;
